storage/imxrt-flash/flashsrv.c

```diff
@@ -379,6 +356,18 @@ static void flashsrv_rawCtl(flash_memory_t *memory, msg_t *msg)
 			odevctl->err = flashsrv_directWrite(memory->fOid.id, memory->parts[partID].pHeader->offset + idevctl->addr, msg->i.data, msg->i.size);
 			break;
 
+		case flashsrv_devctl_directRead:
+			TRACE("imxrt-flashsrv: flashsrv_devctl_directRead, addr: %u, size: %u, id: %u, port: %u.",
+				idevctl->addr + memory->parts[partID].pHeader->offset, msg->o.size, idevctl->oid.id, idevctl->oid.port);
+
+			if (idevctl->addr > memory->parts[idevctl->oid.id].pHeader->size) {
```

Comment:
```suggestion
			if (idevctl->addr >= memory->parts[idevctl->oid.id].pHeader->size) {
```

Reply:
Right. I'll add separate commit in this PR fixing this and other incorrect checks in this switch-case.